library/src/scala/quoted/Quotes.scala

```diff
@@ -53,7 +53,16 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
      *  Emits an error and throws if the expression does not represent a value or possibly contains side effects.
      *  Otherwise returns the value.
      */
-    def valueOrError(using FromExpr[T]): T =
+    @deprecated("Use valueOrThrow", ">3.0.0-RC2")
```

Comment:
If `3.0.0-RC2` gets bumped to `3.0.0` then we should have `>3.0.0`. It would probably be `3.1.0` as we this is a binary incompatible change.

Reply:
Maybe we keep this PR open until we're 100% sure about the plans for RC3? Then I can update this to either `3.0.0-RC2` or `3.0.0`

Reply:
Actually, I just set this to `3.0.0` so that it's ready to go if it's correct. (Other PR feedback addressed.)